src/main/java/org/craftercms/studio/api/v2/dal/security/NormalizedGroup.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2007-2024 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.craftercms.studio.api.v2.dal.security;
+
+import org.apache.commons.lang3.StringUtils;
+
+import static java.util.Objects.requireNonNullElse;
+
+/**
+ * Normalized group to deal with case-insensitive while working with roles/permissions mappings
+ * @param name name of the group
+ */
+public record NormalizedGroup(String name) {
```

Comment:
`NormalizedGroup` and `NormalizedRole` seem identical. Any reason why we would want to keep both?

Reply:
@jmendeza 
In the code, we have many data structures such as:
```
Map<String, Map<NormalizedRole, List<Node>>>
Map<NormalizedRole, List<Node>> siteRoles
Map<NormalizedGroup, List<NormalizedRole>> rolesMap

```
We can have a single class such as `NormalizedString` but it would make the code less readable. And we won't know what type of data the Map key/value is.

Reply:
Got it.